mapper.go

```diff
@@ -24,7 +24,7 @@ import (
 )
 
 var (
-	metricLineRE      = regexp.MustCompile(`^(\*\.|[^*.]+\.)*(\*|[^*.]+)$`)
+	metricLineRE      = regexp.MustCompile(`^(\*\.|[^*.]+\.|\.)*(\*|[^*.]+)$`)
```

Comment:
I'm not sure I understand the problem here. Trimming loses information so I'd prefer to work with the actual data being sent to us.